lib/spack/spack/binary_distribution.py

```diff
@@ -412,7 +414,7 @@ def relocate_package(workdir, allow_root):
     Relocate the given package
     """
     buildinfo = read_buildinfo_file(workdir)
-    new_path = spack.store.layout.root
+    new_path = spack.store.store().layout.root
```

Comment:
This would require the `spack` module to have a symbol called `store` and to know about `spack.store`.  So I'm trying to move away from that.  I do agree this could be nicer.  I'm not 100% sure what the best structure is, though.

I would consider the singletons introduced here to be a stopgap -- they allow lazy loading but they're still essentially global.  I'd like to eventually get to the point where we can instantiate a `Spack` object that just has a `store`, and even have multiple of them in the same process, but we're not there yet.  So I'm not wed to this, but it speeds up init for now.

Reply:
After thinking on it more I'm not even sure there's a problem with this, and also agree that my suggestion may not be the way to handle it if it was a problem (just noting this for future reference).